### ISSUE TITLE
Questions about dataset and training

### ISSUE THREAD

**Issue by Kabnoory** [carla-simulator/imitation-learning#41]
Hey I am trying to implement the paper, but my results are not good. 

1. I noticed that the most of the dataset had a very low steering value even for left or right commands, so was there any data balancing technique used on the training dataset? if yes can you give information about it. 

2. I read in other issues that I need to mask the branches so that you back-propagate only on one branch? I am not sure how to do that, can you give more information on how you did it or some code?

Thanks for your help.

**Comment by felipecode** [comment]
Hey . Kabnoory
1. You are right ! left and right cameras needs to be augmented on the fly. 
This piece of code do the trick
https://gist.github.com/felipecode/0c0bf85c6f3864f7f41c16a78b0416a4

2. I am bit busy right now. I can come back to you later with the code for that.

Cheers

**Comment by Kabnoory** [comment]
Thank you for the code! 
I wanted to ask what is the value of g_conf.AUGMENT_LATERAL_STEERINGS on line 22? 
Also, is there any other augmentations that you apply during training other than this one and the image augmentations?

**Comment by felipecode** [comment]
I think that is all.
It was g_conf.AUGMENT_LATERAL_STEERINGS= 6, the value

**Comment by Kabnoory** [comment]
I think the dataset provided has only 1 camera, so the camera_angle is always 0. So the augment_steering doesn't change anything? is there something I'm missing or is the dataset provided only a portion of the actual dataset?

**Comment by Kabnoory** [comment]
@felipecode Any feedback on whether there is another dataset used in the paper that can be provided or not?

**Comment by jrabary** [comment]
@felipecode @Kabnoory I also check the given dataset and I confirm that every image has camera yaw == 0. But, when you take a close look into each image it seems like there are several cameras. Maybe the dataset was not recorded correctly. @felipecode  Is it possible to check it from your side ?

For example:
![same_view](https://user-images.githubusercontent.com/1025387/45096058-ec381100-b11f-11e8-8d21-84d5fbabe040.jpeg)

**Comment by m-abaas** [comment]
How do we obtain the camera angle anyway, in order to use it in the *lateral_augmentation* function provided by @felipecode ?
